docs/madr/decisions/031-automatic-rechable-services.md

```diff
@@ -0,0 +1,263 @@
+# Automatic reachable services
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/6551
+
+## Context and Problem Statement
+
+With large enough Mesh (more than 1000 services) processing the whole cluster state for each proxy is not possible due to performance reasons (what is the boundary that we're hitting, is it the CPU of the CP / DP, can we have flame graphs showing hot spots?).
```

Comment:
As mentioned before the complexity of this is O(n^2) (n is the number of services). We end up with envoy configs that are multiple MB and CP consuming many cores for small clusters. You can check this by just looking at small clusters with and without reachable services all metrics collapse when using reachable services